src/main/java/io/spokestack/spokestack/nlu/Slot.java

```diff
@@ -0,0 +1,59 @@
+package io.spokestack.spokestack.nlu;
+
+/**
+ * A slot extracted during intent classification.
+ *
+ * <p>
+ * Depending on the NLU service used, slots may be typed; if present, the type
+ * of each slot can be used to cast its raw value to the appropriate class
+ * after NLU has finished.
+ * </p>
+ */
+public final class Slot {
+    private final String slotName;
+    private final Class<?> slotType;
```

Comment:
I don't think you need to track the type of the object - java already gives you a `getClass` method on every object that does this for you

Reply:
For some reason I was thinking that `getClass` called on a slot value declared as `Object` would just return `Class<Object>`. Thanks for reminding me to second-guess that; guess I've been away too long.